tests/e2e/mnist_raycluster_sdk_test.py

```diff
@@ -0,0 +1,152 @@
+from kubernetes import client, config
+import kubernetes.client
+
+import os
+
+from time import sleep
+
+import ray
+
+from torchx.specs.api import AppState, is_terminal
+
+from codeflare_sdk.cluster.cluster import Cluster, ClusterConfiguration
+from codeflare_sdk.job.jobs import DDPJobDefinition
+
+import pytest
+
+from support import random_choice
+
+# Creates a Ray cluster, and trains the MNIST dataset using the CodeFlare SDK.
+# Asserts creation of AppWrapper, RayCluster, and successful completion of the training job.
+# Covers successfull installation of CodeFlare-SDK
+
+class TestMNISTRayClusterSDK:
+    def setup_method(self):
+        # Load the kube config from the environment or Kube config file.
+        config.load_kube_config()
+
+        # Initialize Kubernetes client
+        self.api_instance = client.CoreV1Api()
+        self.custom_api = kubernetes.client.CustomObjectsApi(self.api_instance.api_client)
+
+    def teardown_method(self):
+        if hasattr(self, 'namespace'):
+            self.api_instance.delete_namespace(self.namespace)
+        if hasattr(self, 'configmap'):
+            self.api_instance.delete_namespaced_config_map(self.configmap.metadata.name, self.namespace)
+
+    def test_mnist_ray_cluster_sdk(self):
+        self.create_test_namespace()
+        self.run_mnist_raycluster_sdk()
+
+    def create_test_namespace(self):
+        self.namespace = f"test-ns-{random_choice()}"
+        namespace_body = client.V1Namespace(metadata=client.V1ObjectMeta(name=self.namespace))
+        self.api_instance.create_namespace(namespace_body)
+        return self.namespace
+
+    def run_mnist_raycluster_sdk(self):
+        ray_image = "quay.io/project-codeflare/ray:latest-py39-cu118"
```

Comment:
Can you allow providing custom image using env variable?

Reply:
I added that now, thanks!